lib/reek/smells/nested_iterators.rb

```diff
@@ -37,40 +36,45 @@ def self.default_config
       end
 
       #
-      # Attempts to find the deepest nested iterator and warns if it's depth
+      # Attempts to find the deepest nested iterator and warns if its depth
       # is bigger than our allowed maximum.
       #
       # @return [Array<SmellWarning>]
       #
-      # :reek:TooManyStatements: { max_statements: 6 }
       def inspect(ctx)
-        configure_ignore_iterators(ctx)
-        deepest_iterator = find_deepest_iterator ctx
-        return [] unless deepest_iterator
-        depth = deepest_iterator.depth
-        return [] unless depth > max_nesting(ctx)
+        configure_ignore_iterators ctx
+        violations = find_violations ctx
 
-        [smell_warning(
-          context: ctx,
-          lines: [deepest_iterator.exp.line],
-          message: "contains iterators nested #{depth} deep",
-          parameters: { name: ctx.full_name, count: depth })]
+        violations.group_by(&:depth).map do |depth, group|
+          lines = group.map(&:line)
+          smell_warning(
+            context: ctx,
+            lines: lines,
+            message: "contains iterators nested #{depth} deep",
+            parameters: { name: ctx.full_name, count: depth })
+        end
       end
 
       private
 
       attr_accessor :ignore_iterators
 
+      # Finds deepest iterators that are nested more deeply than allowed.
+      #
+      # @return [Array<Iterator>]
+      #
+      def find_violations(ctx)
+        deepest_iterators = find_deepest_iterators ctx
```

Comment:
But now the naming is confusing (sorry, should have noticed that in my initial review). It's not the "deepest" iterators, it's just all the iterators, isn't it? You check for the depth of the iterators 2 lines below so I'd remove the "deepest" in the local variable name and in the method name.